src/south/sonic/goldstone/south/sonic/portchannel.py

```diff
@@ -26,24 +32,49 @@ def __init__(self, server, change):
 class PortChannelIDHandler(PortChannelChangeHandler):
     def apply(self, user):
         if self.type in ["created", "modified"]:
-            value = self.server.get_default("admin-status")
-            self.server.sonic.set_config_db(
-                self.pid, "admin-status", value, "PORTCHANNEL"
-            )
+            self.mode = self.change.value
+            admin = self.server.get_default("admin-status")
+            mtu = value = self.server.get_default("mtu")
+            if self.mode == "dynamic":
```

Comment:
How about the following to make it cleaner?

```python
            if self.mode == "static":
                self.server.sonic.set_config_db(
                    self.pid, "static", "true", "PORTCHANNEL"
                )
            self.server.sonic.set_config_db(self.pid, "mode", self.mode, "PORTMODE")
            self.server.sonic.set_config_db(
                self.pid, "admin-status", admin, "PORTCHANNEL"
            )
            self.server.sonic.set_config_db(self.pid, "mtu", mtu, "PORTCHANNEL")
```

Reply:
Thank you for your comment. This optimization has been done in the latest commit.